apps/meteor/client/views/room/Header/Omnichannel/QuickActions/hooks/useQuickActions.tsx

```diff
@@ -309,7 +307,10 @@ export const useQuickActions = (): {
 	const canSendTranscriptPDF = usePermission('request-pdf-transcript');
 	const canCloseRoom = usePermission('close-livechat-room');
 	const canCloseOthersRoom = usePermission('close-others-livechat-room');
-	const canPlaceChatOnHold = Boolean(!room.onHold && room.u && !(room as any).lastMessage?.token && manualOnHoldAllowed);
+	const restrictedOnHold = useSetting('Livechat_allow_manual_on_hold_upon_agent_engagement_only');
+	const canRoomBePlacedOnHold = !room.onHold && room.u;
+	const canAgentPlaceOnHold = !room.lastMessage.token;
+	const canPlaceChatOnHold = canRoomBePlacedOnHold && (!restrictedOnHold || canAgentPlaceOnHold);
 
```

Comment:
Saw you removed the usage of the main `allow_manual_on_hold` setting. Could you explain why? 👀 

Now, even when the settings page looks like this, agents are allowed to put chats on hold
![Screenshot from 2023-10-23 13 42 17](https://github.com/RocketChat/Rocket.Chat/assets/11577696/d4e96591-7f59-449d-bbb4-4449195eb128)

Reply:
It was removed as it was not used on any of your suggested code changes, see your quote below, where that setting is not used ....
"I'd do sometihing like:
         const allowUnrestrictedOnHold = useSetting('Livechat_allow_manual_on_hold_upon_agent_engagement_only');
	const canRoomBePlacedOnHold = !room.onHold && room.u;
	const canAgentPlaceOnHold = !room.lastMessage.token;
	const canPlaceChatOnHold = canRoomBePlacedOnHold && (allowUnrestrictedOnHold || canAgentPlaceOnHold)